apps/mitt-konto/src/MittKonto/Main/Views/UserView.purs

```diff
@@ -2,36 +2,50 @@ module MittKonto.Main.UserView where
 
 import Prelude
 
-import Data.Array (snoc, sortBy, (:))
-import Data.Maybe (Maybe(..), maybe)
+import Data.Array (concatMap, snoc, sortBy, (:))
+import Data.Either (Either(..))
+import Data.Maybe (Maybe(..), maybe, fromMaybe)
+import Data.Map as Map
 import Data.Nullable as Nullable
 import Data.String (contains, toUpper)
 import Data.String.Pattern (Pattern(..))
+import Data.Tuple (uncurry)
+import Effect (Effect)
+import Effect.Aff (launchAff_)
+import Effect.Class (liftEffect)
 import MittKonto.Main.UserView.AccountEdit as AccountEdit
 import MittKonto.Main.UserView.IconAction as IconAction
 import MittKonto.Main.Elements as Elements
 import MittKonto.Main.Helpers as Helpers
 import MittKonto.Main.Types as Types
 import MittKonto.Main.UserView.Subscription (subscription) as Subscription
+import KSF.AsyncWrapper (loadingSpinner)
 import KSF.Api.Subscription (isSubscriptionCanceled, isSubscriptionExpired) as Subscription
+import KSF.Paper as Paper
 import KSF.Sentry as Sentry
 import KSF.User (User)
+import KSF.User as User
 import KSF.User.Cusno as Cusno
+import Persona (Newsletter, NewsletterSubscription)
 import React.Basic (JSX)
 import React.Basic.DOM as DOM
+import React.Basic.Events (handler)
+import React.Basic.DOM.Events (targetValue, capture_)
 import React.Basic.Hooks as React
 import Routing.PushState (PushStateInterface)
 
 foreign import images :: { subscribe :: String }
 foreign import _encodeURIComponent :: String -> String
 
 -- | User info page with profile info, subscriptions, etc.
-userView :: PushStateInterface -> Types.Self -> Sentry.Logger ->  JSX -> User -> JSX
-userView router { state: { now, news } } logger profileComponent user = React.fragment
+userView :: PushStateInterface -> Types.Self -> Sentry.Logger ->  JSX -> ((Types.State -> Types.State) -> Effect Unit) -> User -> JSX
```

Comment:
It's probably a bigger refactor best left for later but this is starting to look like it'd be better off as a proper React component.

Reply:
Yeah, I agree. I considered doing that here as well, but since the Purescript way of creating React components is pretty unwieldy, I just decided to do it like this.